test/models/architecture_test.rb

```diff
@@ -74,4 +74,45 @@ class ArchitectureTest < ActiveSupport::TestCase
     architecture = Architecture.new :name => "../../etc/shadow"
     assert_equal "etc-shadow", architecture.bootfilename_efi
   end
+
+  test "create with multi names" do
+    valid_name_list.each do |name|
+      architecture = FactoryBot.build(:architecture, :name => name)
+      assert architecture.valid?, "Validation failed for create with valid name: '#{name}' length: #{name.length})"
+      assert_equal architecture.name, name
+    end
+  end
+
+  test "should not create with invalid names" do
+    invalid_name_list.each do |name|
+      architecture = FactoryBot.build(:architecture, :name => name)
+      refute architecture.valid?, "Validation succeeded for create with invalid name: '#{name}' length: #{name.length})"
+      assert_includes architecture.errors.keys, :name
+    end
+  end
+
+  test "update with multi names" do
+    architecture = FactoryBot.create(:architecture)
+    valid_name_list.each do |new_name|
+      architecture.name =  new_name
+      assert architecture.valid?, "Validation failed for update with valid name: '#{new_name}' length: #{new_name.length})"
+      assert_equal architecture.name, new_name
+    end
+  end
+
+  test "should not update with invalid names" do
+    architecture = Architecture.first
+    invalid_name_list.each do |name|
+      architecture.name = name
+      refute architecture.valid?, "Validation succeeded for update with invalid name: '#{name}' length: #{name.length})"
```

Comment:
`refute_valid` can be used here.

Reply:
same here as assert_valid,  we are going to loose the failure message as refute_valid does not accept message and  we need that messages for debugging as we have a loop over different type and length of strings.